### DISCUSSION TITLE
怎样在新建注释时调整条目标签？

### DISCUSSION THREAD

**Discussion by Shalloater** [windingwind/zotero-actions-tags#246] [Q&A]
求助各位！我的事件设置为新建注释：
![image](https://github.com/windingwind/zotero-actions-tags/assets/78467381/f0bc2e13-1194-4b0d-8afd-0594d5b5ca81)
我希望在我新建注释的时候，将标签替换为/read，使用脚本如下：
```js
// Replace tags
// Modify this line 👇
const tagsToReplace = ["/unread"]; // an array of tags to remove
const tagsToMaintain = ["/stuck", "/organize", "/finish"]; 

// Modify this line 👇
const tagToAdd = "/read";

// loop through the tags to replace
for (let single_tag of tagsToReplace) {
	// check if the item has the tag
	if (item.getTags().some(obj => obj.tag === single_tag)) {
		item.removeTag(single_tag);
		item.addTag(tagToAdd);
		return `[Replace Tags] replace tag ${single_tag} to ${tagToAdd} on ${item.getField("title")}`;
	}
}

for (let single_tag of tagsToMaintain) {
	// check if the item has the tag
	if (item.getTags().some(obj => obj.tag === single_tag)) {
		return;
	}
}

if (item.getTags().some(obj => obj.tag === tagToAdd)) {
	return;
}

item.addTag(tagToAdd);
return `Add tag ${tagToAdd} on ${item.getField("title")}`;
```
但是这样会将标签加在新建的注释上，无法改变条目的标签，我想要改变父条目的标签，就加入了寻找父项目的语句，并把`item`全部替换为了`parentItem`：
```js
if (!item) return;
let parentItem = Zotero.Items.getTopLevel(item)[0];
```
但运行会报错：items.map is not a function
![image](https://github.com/windingwind/zotero-actions-tags/assets/78467381/285f5a44-7d34-4199-a2d7-05e8c3377c61)
请问这个问题怎样解决？

**Comment by ramonmi** [comment]
遇到了同样的问题，请问有其他解决思路吗

**Reply by Shalloater** [reply]
我现在是将事件改为“被添加注释”就可以解决

**Comment by ramonmi** [comment]
可以尝试将 let parentItem = Zotero.Items.getTopLevel(item)[0]; 改为 let parentItem = Zotero.Items.getTopLevel([item])[0]; See https://github.com/windingwind/zotero-actions-tags/discussions/340
